### PR TITLE
Make script editor translatable

### LINKED ISSUE #3848: Make the script editor translatable
All text in the Home Assistant UI should be translated.

The [script editor](https://github.com/home-assistant/home-assistant-polymer/tree/dev/src/panels/config/script) is not translatable yet.
- Create new keys in `src/translations/en.json` with the text that is currently hard coded in the source.
Try to follow the same logic that is present, and give every section in Home Assistant it's own object.
If you would place the text in
```json
  "ui": {
    "panel": {
      "example": {
        "text": "Here goes the text",
      },
    },
  },
```
your key would be: `"ui.panel.example.text`

Polymer Elements need a mixin: `LocalizeMixin(PolymerElement)`, Lit Elements work without a mixin. 

- Replace the text in the code with:
  - Lit element: `${this.hass.localize("ui.panel.example.text")}`
  - Polymer: `[[localize('ui.panel.history.showing_entries')]]`

_(This is part of https://github.com/home-assistant/home-assistant-polymer/issues/3429)_

### PR DESCRIPTION
**Pull request by marcelbrueckner**
Introduces translatable keys for script editor analogous to automation editor's translation.

Resolves #3848.

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @marcelbrueckner, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant-polymer#3866).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/panels/config/script/ha-script-editor.js [hunk 1]
```diff
@@ -254,7 +259,7 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
     if (
       this.dirty &&
       // eslint-disable-next-line
-      !confirm("You have unsaved changes. Are you sure you want to leave?")
+      !confirm("[[localize('ui.panel.config.script.editor.unsaved_confirm')]]")
```
Comment (iantrich) [inline src/panels/config/script/ha-script-editor.js]: This one is likely more general than script editor

Comment (marcelbrueckner) [reply src/panels/config/script/ha-script-editor.js]: Should I put it below `ui.common` then? I have oriented myself on the automation editor's translation. In addition, basically every other section has it's own strings, too (just search for `delete_confirm`or `confirm_delete` in `en.json`).

Comment (iantrich) [reply src/panels/config/script/ha-script-editor.js]: Interesting...we should clean that up. We're just making more work for those who need to translate. I think maybe ui.panel.config.editor.common?

Comment (marcelbrueckner) [reply src/panels/config/script/ha-script-editor.js]: Alright. Will use `ui.panel.config.common.editor`, though, to keep up with the scheme `ui.panel.config.SECTION.editor`.

File: src/panels/config/script/ha-script-picker.ts [hunk 2]
```diff
@@ -97,7 +113,9 @@ class HaScriptPicker extends LitElement {
     const script = ev.currentTarget.script as HassEntity;
     await triggerScript(this.hass, script.entity_id);
     showToast(this, {
-      message: `Triggered ${computeStateName(script)}`,
+      message: `${this.hass.localize(
+        "ui.dialogs.notification_toast.triggered"
+      )} ${computeStateName(script)}`,
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-picker.ts]: Let's give `computeStateName(script)` as variable to lokalize, so the order can be changed by the translator:

```js
${this.hass.localize("ui.dialogs.notification_toast.triggered", "name", computeStateName(script))}
```

The string in `en.json`would be:
`Triggered {name}`

File: src/panels/config/script/ha-script-editor.js [hunk 3]
```diff
@@ -100,7 +100,10 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
             <ha-paper-icon-button-arrow-prev
               on-click="backTapped"
             ></ha-paper-icon-button-arrow-prev>
-            <div main-title>Script [[computeName(script)]]</div>
+            <div main-title>
+              [[localize('ui.panel.config.script.caption')]]
+              [[computeName(script)]]
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-editor.js]: Let's give `computeName(script)` as variable to lokalize, so the order can be changed by the translator:

```js
[[localize('ui.panel.config.script.caption'), 'name', computeName(script)]]
```

The string in `en.json`would be:
`Script {name}`

File: src/panels/config/script/ha-script-editor.js [hunk 4]
```diff
@@ -281,7 +286,9 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
   }
 
   async _delete() {
-    if (!confirm("Are you sure you want to delete this script?")) {
+    if (
+      !confirm("[[localize('ui.panel.config.script.editor.delete_confirm')]]")
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-editor.js]: Remove the "", it is no longer a string
```suggestion
      !confirm([[localize('ui.panel.config.script.editor.delete_confirm')]])
```

File: src/panels/config/script/ha-script-editor.js [hunk 5]
```diff
@@ -254,7 +259,7 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
     if (
       this.dirty &&
       // eslint-disable-next-line
-      !confirm("You have unsaved changes. Are you sure you want to leave?")
+      !confirm("[[localize('ui.panel.config.common.editor.confirm_unsaved')]]")
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-editor.js]: Remove the ""

File: src/panels/config/script/ha-script-editor.js [hunk 6]
```diff
@@ -232,7 +235,9 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
           this._updateComponent();
         },
         () => {
-          alert("Only scripts inside scripts.yaml are editable.");
+          alert(
+            "[[localize('ui.panel.config.script.editor.load_error_not_editable')]]"
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-editor.js]: Remove the ""

File: src/panels/config/script/ha-script-editor.js [hunk 7]
```diff
@@ -244,7 +249,7 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
     }
     this.dirty = false;
     this.config = {
-      alias: "New Script",
+      alias: "[[localize('ui.panel.config.script.editor.default_name')]]",
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-editor.js]: Remove the ""

File: src/panels/config/script/ha-script-editor.js (hunk 4 above)
Comment (bramkragten) [reply src/panels/config/script/ha-script-editor.js]: And use `this.hass.localize`, as it is not used inside the Polymer render.

File: src/panels/config/script/ha-script-editor.js (hunk 6 above)
Comment (marcelbrueckner) [reply src/panels/config/script/ha-script-editor.js]: If I remove the quotes, `npm run lint` throws following errors:

```
  238:17  error  Insert `[`                                                                                                                                                 prettier/prettier
  239:14  error  Replace `[localize('ui.panel.config.script.editor.load_error_not_editable')]]` with `localize("ui.panel.config.script.editor.load_error_not_editable")],`  prettier/prettier
  239:15  error  'localize' is not defined                                                                                                                                  no-undef
  240:11  error  Insert `]`                                                                                                                                                 prettier/prettier
```

After trying to fix the new linting issues, the error `'localize' is not defined` still remains.

Comment (bramkragten) [reply src/panels/config/script/ha-script-editor.js]: Yes, because it is not inside the render method of Polymer, use `this.hass.localize`

Comment (marcelbrueckner) [reply src/panels/config/script/ha-script-editor.js]: Thank you very much. I think I've got the point now.

**Review by bramkragten — APPROVED** [review]
Great thanks! 🦁

**Pull request merged**